ngraph/core/include/openvino/core/node.hpp

```diff
@@ -513,6 +514,7 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
     std::shared_ptr<ngraph::op::util::OpAnnotations> m_op_annotations;
     OPENVINO_SUPPRESS_DEPRECATED_END
     std::map<std::string, std::shared_ptr<Variant>> m_rt_info;
+    mutable std::mutex m_mutex;
```

Comment:
what if we have thousands of nodes- that would be too bad (afaik only 16M mutexes are allowed in Linux)... also having a mutex per node seems to be VERY redundant